### ISSUE TITLE
Can't find Fortran headers

### ISSUE THREAD

**Issue by downloadico** [spack/spack#11588]
I ran `spack install abinit+hdf5` and it failed with:
```console
==> Error: NoHeadersError: Unable to locate netcdf-fortran headers in <netcdf-fortran's include directory>
```

### Steps to reproduce the issue

```console
$ spack install abinit+hdf5
```

### Error Message

```console
==> Error: NoHeadersError: Unable to locate netcdf-fortran headers in <netcdf-fortran's include directory>
```

### Information on your system

This looks to me to be because Spack doesn't understand that not all header filenames end with `.h`!  I'm pretty sure the problem is related to `find_headers` in `lib/spack/llnl/util/filesystem.py`.

**Comment by adamjstewart** [comment]
> not all header filenames end with `.h`!

We currently search for `.h`, `.hpp`, and `.mod`. Is there anything else we should add? This should be a really easy fix.

**Comment by adamjstewart** [comment]
My install finally finished. It looks like we can add `.inc` to the list. However, the directory already includes `.mod` files, so it shouldn't be failing like this. I'll try to investigate further.

**Comment by adamjstewart** [comment]
I was unable to reproduce this problem. Abinit successfully added the following argument to configure:
```
--with-netcdf-incs=-I/Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netcdf-fortran-4.4.4-w775s4x3xzbfgpg256ebtzo5zrqvfcrf/include
```
Question: are you using the latest version of Spack? I have a feeling that the bug you are encountering was fixed by #10798 (merged a few months ago). 

If not, @alalazo might want to take a look at this, as he's handled a lot of updates to `abinit` over the years. I did notice a few minor bugs in the package. For example, we add an `hdf5` dependency only when `+hdf5+mpi` but we use this dependency in configure_args when `+hdf5~mpi`. We also query `hdf5:hl` but we don't explicitly depend on `hdf5+hl`. This is also true for the other dependencies.
